sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -0,0 +1,550 @@
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+import abc
+import json
+from typing import TYPE_CHECKING, Optional, Any, Union
+
+from ..exceptions import HttpResponseError, DecodeError
+from . import PollingMethod
+
+if TYPE_CHECKING:
+    from azure.core.pipeline import PipelineResponse
+    from azure.core.pipeline.transport import (
+        HttpResponse,
+        AsyncHttpResponse,
+        HttpRequest,
+    )
+
+    ResponseType = Union[HttpResponse, AsyncHttpResponse]
+    PipelineResponseType = PipelineResponse[HttpRequest, ResponseType]
+
+
+try:
+    ABC = abc.ABC
+except AttributeError:  # Python 2.7, abc exists, but not ABC
+    ABC = abc.ABCMeta("ABC", (object,), {"__slots__": ()})  # type: ignore
+
+
+FINISHED = frozenset(["succeeded", "canceled", "failed"])
+FAILED = frozenset(["canceled", "failed"])
+SUCCEEDED = frozenset(["succeeded"])
+
+
+def finished(status):
+    if hasattr(status, "value"):
+        status = status.value
+    return str(status).lower() in FINISHED
+
+
+def failed(status):
+    if hasattr(status, "value"):
+        status = status.value
+    return str(status).lower() in FAILED
+
+
+def succeeded(status):
+    if hasattr(status, "value"):
+        status = status.value
+    return str(status).lower() in SUCCEEDED
+
+
+class BadStatus(Exception):
+    pass
+
+
+class BadResponse(Exception):
+    pass
+
+
+class OperationFailed(Exception):
+    pass
+
+
+def _as_json(response):
+    # type: (ResponseType) -> dict
+    """Assuming this is not empty, return the content as JSON.
+
+    Result/exceptions is not determined if you call this method without testing _is_empty.
+
+    :raises: DecodeError if response body contains invalid json data.
+    """
+    try:
+        return json.loads(response.text())
+    except ValueError:
+        raise DecodeError("Error occurred in deserializing the response body.")
+
+
+def _raise_if_bad_http_status_and_method(response):
+    # type: (ResponseType) -> None
+    """Check response status code is valid.
+
+    Must be 200, 201, 202, or 204.
+
+    :raises: BadStatus if invalid status.
+    """
+    code = response.status_code
+    if code in {200, 201, 202, 204}:
```

Comment:
Do we actually have LROs that return 204?

Reply:
Yes, some DELETE
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#delete-resource-asynchronously

Reply:
Gah, RPC.